### ISSUE TITLE
Add list-of-vignettes feature to all vignettes

### ISSUE THREAD

**Issue by bryanhanson** [cbeleites/hyperSpec#261]
Currently, the intro vignette `hyperSpec.Rmd` has code to insert a file `list-of-vignettes.md` as well as internal links that point to the inserted code.  This greatly reduces the need to edit the files if the name of the vignettes change, their links change or more vignettes are added, as everything is centralized.

These features should be added to the other vignettes when they are reviewed:

- [ ] Fixed in `baseline.Rmd`
- [ ] Fixed in `chondro.Rmd`
- [ ] Fixed in `flu.Rmd`
- [ ] Fixed in `fileio.Rmd`
- [ ] Fixed in `plotting.Rmd`

**Comment by GegznaV** [comment]
The hySpc.ggplot2 and the other vignettes in hySpc packages  should also be
included in this list.

2020 rugp. 17, pr 22:36, Bryan Hanson <notifications@github.com> rašė:

**Comment by bryanhanson** [comment]
Good point.  Probably we should not edit the ones here in `hyperSpec` that have been moved to other repos, and instead edit them there, eventually deleting them here.  What do you think @GegznaV ?

**Comment by bryanhanson** [comment]
Maybe I should move this issue to the `r-hyperspec.githhub.io` issue list?

**Comment by eoduniyi** [comment]
@bryanhanson Let me make sure I got this straight: vignettes should point to each other via the `list-of-vignettes` and internal links. 

`hySpc.read.txt(vignette) <-> hyperSpec(vignette)`

**Comment by bryanhanson** [comment]
Did @GegznaV fix this?  One of his commits mentioned cross-references but that was probably section references(?).

But generally, mention of a different vignette (vig 2) somewhere deep in vig1 is marked by an internal link like this:

```
In addition, [`laser`](#list-of-vignettes)...
```

which when clicked takes the reader to the top of vig1 where they see the colored box with information about all vignettes that are available.  This colored box is placed in (eventually) every vig  by this code near the very top:

```
{r, echo = FALSE, results = "asis"}
res <- knitr::knit_child("list-of-vignettes.md", quiet = TRUE)
cat(res, sep = '\n')
```
```
It's operational in `hyperSpec.Rmd`

**Comment by eoduniyi** [comment]
> Did @GegznaV fix this? One of his commits mentioned cross-references but that was probably section references(?).
> 
> But generally, mention of a different vignette (vig 2) somewhere deep in vig1 is marked by an internal link like this:
> 
> ```
> In addition, [`laser`](#list-of-vignettes)...
> ```
> 
> which when clicked takes the reader to the top of vig1 where they see the colored box with information about all vignettes that are available. This colored box is placed in (eventually) every vig by this code near the very top:
> 
> ```
> {r, echo = FALSE, results = "asis"}
> res <- knitr::knit_child("list-of-vignettes.md", quiet = TRUE)
> cat(res, sep = '\n')
> ```
> 
> ```
> It's operational in `hyperSpec.Rmd`
> ```

@bryanhanson have you seen issue #270

**Comment by bryanhanson** [comment]
OK, getting up to speed at >40 messages...  270 is a different issue from the internal links.  I just checked and the internal links work in the intro vignette, but there are some cross references that are broken as reported in 270.  Are we on the same page?   I'm a little confused.

**Comment by GegznaV** [comment]
Let's focus on "*Add list-of-vignettes feature to all vignettes*" here and open separate issues for all other topics.
